### DISCUSSION TITLE
Customize indeterminate ProgressBar

### DISCUSSION THREAD

**Discussion by akordowski** [spectreconsole/spectre.console#1282] [General]
As I could not find any example, can anyone tell me how I can customize the indeterminate ProgressBar, e.g. change the dashed line to a solid one or change the thickness? Thanks in advance.

**Comment by akordowski** [comment]
I also tried to change the spinner but it always shows the default one. What am I making wrong?

![grafik](https://github.com/spectreconsole/spectre.console/assets/9746197/3bae55d2-552c-4664-b792-16e8529c962e)

```
AnsiConsole
    .Progress()
    .Columns(new ProgressColumn[]
    {
        new TaskDescriptionColumn(),
        new ProgressBarColumn(),
        new SpinnerColumn
        {
            Spinner = Spinner.Known.Dots2
        }
    })
    .Start(ctx =>
    {
        var task = ctx.AddTask("Description");
        task.IsIndeterminate = true;

        while (!ctx.IsFinished)
        {
        }
    });
```

**Reply by patriksvensson** [reply]
It's using a fallback spinner since your terminal output encoding isn't a Unicode one.

**Reply by akordowski** [reply]
Ahh, good to know. Thank you!

Do you maybe have an hint how to customize the ProgressBar?

**Reply by patriksvensson** [reply]
Here is the code for the progress bar:

* https://github.com/spectreconsole/spectre.console/blob/main/src/Spectre.Console/Live/Progress/Columns/ProgressBarColumn.cs
* https://github.com/spectreconsole/spectre.console/blob/main/src/Spectre.Console/Widgets/ProgressBar.cs

**Comment by akordowski** [comment]
I figured out that Windows don't have activated the UTF-8 support by default. I found [this article ](https://akr.am/blog/posts/using-utf-8-in-the-windows-terminal) which helped me to activate the UTF-8 support. Now the Windows terminal shows the ProgressBar as a solid line and the spinner is also displayed correctly.

**Reply by patriksvensson** [reply]
We also have an article explaining this here: https://spectreconsole.net/best-practices
